internal/ent/hooks/standard.go

```diff
@@ -5,13 +5,71 @@ import (
 
 	"entgo.io/ent"
 
+	"github.com/theopenlane/entx"
+	"github.com/theopenlane/iam/auth"
 	"github.com/theopenlane/iam/fgax"
 
 	"github.com/theopenlane/core/internal/ent/generated"
+	"github.com/theopenlane/core/internal/ent/generated/control"
 	"github.com/theopenlane/core/internal/ent/generated/hook"
+	"github.com/theopenlane/core/internal/ent/generated/privacy"
 	"github.com/theopenlane/core/internal/ent/generated/standard"
 )
 
+// HookStandardDelete cascades the deletion of all controls that have a standard linked/connected
+func HookStandardDelete() ent.Hook {
+	return hook.On(func(next ent.Mutator) ent.Mutator {
+		return hook.StandardFunc(func(ctx context.Context, m *generated.StandardMutation) (generated.Value, error) {
+
+			if !entx.CheckIsSoftDelete(ctx) {
+				return next.Mutate(ctx, m)
+			}
+
+			if !auth.IsSystemAdminFromContext(ctx) {
+				return next.Mutate(ctx, m)
+			}
+
+			id, _ := m.ID()
+			var err error
+
+			ctx = privacy.DecisionContext(ctx, privacy.Allow)
+
+			retrievedStandard, err := m.Client().Standard.Get(ctx, id)
```

Comment:
the user in context should have access or this should fail anyways? Why do we need the privacy allow? 

We should also just grab the fields we need, not all of them. 
```suggestion
			retrievedStandard, err := m.Client().Standard.Query().Where(standard.ID(id)).Select(standard.FieldSystemOwned,standard.IsPublic).Only(ctx)
```

Another thing, I think we shouldn't allow the delete of public standards, so can you add if retrievedStandard.IsPublic - return error public statndard not allowed to be deleted.